src/core/border_router/routing_manager.cpp

```diff
@@ -3327,11 +3332,61 @@ void RoutingManager::RsSender::HandleTimer(void)
 RoutingManager::PdPrefixManager::PdPrefixManager(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mEnabled(false)
+    , mIsRunning(false)
     , mTimer(aInstance)
 {
     mPrefix.Clear();
 }
 
+void RoutingManager::PdPrefixManager::Start(void)
+{
+    Dhcp6PdState oldState = GetState();
+
+    VerifyOrExit(!mIsRunning);
+    mIsRunning = true;
+    EvaluateStateChange(oldState);
+
+exit:
+    return;
+}
+
+void RoutingManager::PdPrefixManager::Stop(void)
+{
+    Dhcp6PdState oldState = GetState();
+
+    VerifyOrExit(!mIsRunning);
+    mIsRunning = false;
+    EvaluateStateChange(oldState);
+
+exit:
+    return;
+}
+
+void RoutingManager::PdPrefixManager::EvaluateStateChange(Dhcp6PdState aOldState)
+{
+    Dhcp6PdState newState = GetState();
+
+    VerifyOrExit(aOldState != newState);
+    switch (newState)
+    {
+    case Dhcp6PdState::kDhcp6PdStateDisabled:
```

Comment:
We don't need `Dhcp6PdState::` here and below (since it is normal `enum`). 

Suggest removing it... Some older toolchains give warning/error if a normal `enum` is used as a namespace.